x-pack/plugins/observability_solution/inventory/common/entities.ts

```diff
@@ -117,3 +121,9 @@ export interface ContainerEntity extends BaseEntity {
 }
 
 export type Entity = ServiceEntity | HostEntity | ContainerEntity;
+
+export type EntityGroup = {
+  count: number;
+} & {
+  [Field in keyof Entity]?: Entity[Field];
```

Comment:
We will remove the `Entity` type very soon. I think it's better if you change it to string.